test/project_types/script/layers/infrastructure/script_project_repository_test.rb

```diff
@@ -353,63 +393,180 @@ def hash_except(config, *keys)
-        assert_equal script_json_content["configuration"].to_json, script_json.configuration.to_json
-        assert_equal script_json_content["configuration"].to_json, file_content["configuration"].to_json
+        assert_equal script_config_content["version"], script_config.version
+        assert_equal script_config_content["version"], file_content["version"]
+        assert_equal script_config_content["configuration"].to_json, script_config.configuration.to_json
+        assert_equal script_config_content["configuration"].to_json, file_content["configuration"].to_json
+      end
+    end
+  end
+
+  describe "ScriptConfigYmlRepository" do
+    let(:instance) { Script::Layers::Infrastructure::ScriptProjectRepository::ScriptConfigYmlRepository.new(ctx: ctx) }
+    let(:version) { "2" }
+    let(:title) { "title" }
+    let(:content) { { "version" => version, "title" => title }.to_yaml }
+
+    describe "get" do
+      subject { instance.get }
+
+      describe "when file does not exist" do
+        it "returns nil" do
+          assert_nil(subject)
+        end
+      end
+
+      describe "when file exists" do
+        before do
+          File.write(script_config_filename, content)
+        end
+
+        describe "when content is invalid yaml" do
+          let(:content) { "*" }
+
+          it "raises InvalidScriptConfigYmlDefinitionError" do
+            assert_raises(Script::Layers::Infrastructure::Errors::InvalidScriptConfigYmlDefinitionError) { subject }
+          end
+        end
+
+        describe "when content is not a hash" do
+          let(:content) { "" }
+
+          it "raises InvalidScriptConfigYmlDefinitionError" do
+            assert_raises(Script::Layers::Infrastructure::Errors::InvalidScriptConfigYmlDefinitionError) { subject }
+          end
+        end
+
+        describe "when content is missing fields" do
+          let(:content) { {}.to_yaml }
+
+          it "raises MissingScriptConfigYmlFieldError" do
+            assert_raises(Script::Layers::Infrastructure::Errors::MissingScriptConfigYmlFieldError) { subject }
+          end
+        end
+
+        describe "when content is valid yaml" do
+          it "returns the entity" do
+            assert_equal version, subject.version
+          end
+        end
+      end
+    end
+
+    describe "update_or_create" do
+      let(:new_title) { "new title" }
+      subject { instance.update_or_create(title: new_title) }
+
+      describe "when file does not exist" do
+        it "creates a ScriptConfig" do
+          refute_nil(subject)
+          assert_equal(new_title, subject.title)
+          assert_equal(version, subject.version)
+        end
+      end
+
+      describe "when file does exist" do
+        before do
+          File.write(script_config_filename, content)
+        end
+
+        it "updates the ScriptConfig" do
+          assert_equal(new_title, subject.title)
+          assert_equal(version, subject.version)
```

Comment:
Should we check the other fields here too?

Reply:
Good call. I added some more checks here, plus some checks to ensure that the files were updated correctly